### ISSUE TITLE
XAML designer does not work on dotnet core 2.0

### ISSUE THREAD

**Issue by vitidev** [AvaloniaUI/Avalonia#1134]
Many errors like:

> State Error NU1605	Detected package downgrade: System.Collections from 4.3.0 to 4.0.11. Reference the package directly from the project to select a different version. 
 AvaloniaCore (>= 1.0.0) -> Avalonia (>= 0.5.0) -> Microsoft.Extensions.DependencyModel (>= 1.1.0) -> Microsoft.DotNet.PlatformAbstractions (>= 1.1.0) -> System.IO.FileSystem (>= 4.0.1) -> runtime.win.System.IO.FileSystem (>= 4.3.0) -> System.Collections (>= 4.3.0) 
 AvaloniaCore (>= 1.0.0) -> Avalonia (>= 0.5.0) -> Microsoft.Extensions.DependencyModel (>= 1.1.0) -> Microsoft.DotNet.PlatformAbstractions (>= 1.1.0) -> System.Collections (>= 4.0.11)	AvaloniaCore

**Comment by kekekeks** [comment]
Our previewer can *only* work with full .NET, unfortunately. The work on [more portable one](https://github.com/AvaloniaUI/Avalonia/issues/900) is currently [in progress](https://github.com/AvaloniaUI/Avalonia/pull/1105).

To use current previewer you need to either use multitargeting (i. e. set add both `net461` and `netcoreapp2.0` to `TargetFrameworks` property) or have everything in a netstandard library and reference it from netcoreapp2.0 and `net461` apps.

**Comment by vitidev** [comment]
Very strange. I have Win 7 x64+VS2017(+all updates) with [extension ](https://marketplace.visualstudio.com/items?itemName=AvaloniaTeam.AvaloniaforVisualStudio)+ full .NET

Yesterday, after changing TargetFrameworks to `netcoreapp2.0;net461`, the designer did not work, but today it works. Mystic.

But still the error list is full of "NU1605 Detected package downgrade" and I cannot update nuget packages.

ps: I found a solution. I install Newtonsoft.Json from nuget, who causes multiple uninstall or update "System.*"  and after that I no longer see any NU1605 errors.

**Comment by kekekeks** [comment]
You are seeing errors about package downgrade because nuget's version resolution was designed by morons. See "Nearest wins" section [there](https://docs.microsoft.com/en-us/nuget/consume-packages/dependency-resolution#dependency-resolution-rules).

TL;DR: NuGet can and *will* select a package version that does *not* satisfy requirements of some packages in dependency tree.

So you need to ensure that you are using correct versions.

**Comment by jkoritzinsky** [comment]
Now that our designer supports . net core 2.0 apps I'm going to close this issue.

**Comment by pamela032709** [comment]
are they fixing this issue?

**Comment by vitidev** [comment]
@pamela032709 Is it still actual?

2 weeks ago I developed an application (.core 2.2 / 3) and the designer worked.

**Comment by pamela032709** [comment]
Yes, less than a week . is just not there . Do I need to do something to be able  to see it ?

**Comment by kekekeks** [comment]
1) Install the latest version of Visual Studio 2019
2) Install the latest version of [our VS extension](https://marketplace.visualstudio.com/items?itemName=AvaloniaTeam.AvaloniaforVisualStudio)
3) Install the latest version of Avalonia from nuget (currently 0.8.1)
4) Most importantly: build your project first, intellisense and previewer require the compiled binary
